test/python/circuit/test_parameters.py

```diff
@@ -139,6 +140,41 @@ def test_get_parameters(self):
         self.assertIs(theta, next(iter(vparams)))
         self.assertEqual(rxg, vparams[theta][0][0])
 
+    def test_get_parameters_by_index(self):
+        """Test getting parameters by index"""
+        x = Parameter('x')
+        y = Parameter('y')
+        z = Parameter('z')
+        v = ParameterVector('v', 3)
+        qc = QuantumCircuit(1)
+        qc.rx(x, 0)
+        qc.rz(z, 0)
+        qc.ry(y, 0)
+        qc.u(*v, 0)
+        self.assertEqual(x, qc._parameter_table[0])
+        self.assertEqual(y, qc._parameter_table[2])
+        self.assertEqual(z, qc._parameter_table[1])
+        for i, vi in enumerate(v):
+            self.assertEqual(vi, qc._parameter_table[3+i])
```

Comment:
Would it be better to test these on `.parameters` instead of `._parameter_table`?